lib/cc/analyzer/formatters/templates/html.erb

```diff
@@ -1,53 +1,637 @@
-
-    <% @issues_by_path.each do |path, file_issues| %>
-      <hr>
-      <h4><a name='<%= path %>'><%= path %></a> (<%= pluralize(file_issues.size, "issue") %>)</h4>
-      <hr>
-      <% file_issues.each do |issue| %>
-        <table class="table table-striped">
-          <tr>
-            <td>Description</td><td><%= issue["description"] %></td>
-          </tr>
-          <tr>
-            <td>Location</td><td><%= path %>:<b><%= issue["location"] %></td>
-          </tr>
-          <tr>
-            <td>Engine</td><td><%= issue["engine_name"] %></td>
-          </tr>
-          <tr>
-            <td>Category</td><td><%= issue["categories"].join(",") %></td>
-          </tr>
-          <tr>
-            <td>Remediation points</td><td><%= issue["remediation_points"] %></td>
-          </tr>
-        </table>
-        <% if issue["content"] %>
-          <br/>
-          <button data-toggle="collapse"
-                  data-target='#readup<%= issue['fingerprint'] %>'>
-                  Toggle Readup
-          </button>
-          <div id='readup<%= issue["fingerprint"] %>' class='collapse'>
-            <hr>
-            <%= issue["content"]["body"] %>
-            <hr>
+    <header id="top">
+      <div class="container">
+        <a id="logo" href="https://codeclimate.com/">Code Climate</a>
+        <h1><%= h project_name %></h1>
+      </div>
+    </header>
+    <div class="container">
+      <nav>
+        <ul>
+          <li>
+            Issues
+          </li>
+        </ul>
+      </nav>
+      <div id="main-container">
+        <%- if issues.any? %>
+          <div class="issue-filters">
+            <label for="category-filter">Category</label>
+            <select id="category-filter">
+              <option value="">All Categories</option>
+              <%- issues.categories.each do |category| -%>
+              <option value="<%= param category %>"><%= category %></option>
+              <%- end -%>
+            </select>
+            <label for="engine-filter">Engine</label>
+            <select id="engine-filter">
+              <option value="">All Engines</option>
+              <%- issues.engines.each do |engine| -%>
+              <option value="<%= param engine %>"><%= engine %></option>
+              <%- end -%>
+            </select>
+          </div>
+          <ul id="smells">
+            <%- issues.each do |issue| -%>
+            <li
+              data-categories="<%= params issue.categories %>"
+              data-engine="<%= param issue.engine_name %>">
+              <h2><%= h issue.description %></h2>
+              <div class="code">
+                <pre class="line-numbers <%= issue.source.syntaxes.map {|l| "language-#{l}" }.join ' ' %>"
+                     data-line="<%= issue.location %>"
+                     data-start="<%= issue.location.start %>"
+                     data-line-offset="<%= issue.location.line_offset %>"><code><%=
```

Comment:
`data-line-offset` is used by Prism's line highlight plugin. The one that gives different background to affected lines.

The removed code was based on that plugin and used it for the same calculations the plugin does. Removed code was used to properly position the whole source within a small window showing relevant piece of code. And it was removed since we don't do that any more (we don't position the code, we display the whole piece available for Prism now).